### ISSUE TITLE
SoftConstraint subclass of Constraint

### ISSUE THREAD

**Issue by arcondello** [dwavesystems/dwavebinarycsp#73]
**Application**
Right now all the constraints are soft constraints when converted to BQMs. It would be good though to formalize that and allow users to specify the strength of their constraints

**Comment by mareksubocz** [comment]
That's right! It would allow for a high-level way of setting truly soft constraints.
